### PR TITLE
Fix toolbar icon and text color in white theme

### LINKED ISSUE #3042: Polish white theme: make app bar text and icons white
This is how the white theme looks like:

As you can see text and icons in the title bar are black, even though they are on the exact same red background as in the dark theme. Black on red looks strange, so text and icons should be white, as in the Retry button

### PR DESCRIPTION
**Pull request by IAmSSH**
#### What is it?
- [ ] Bug fix (user facing)
- [x] Feature (user facing)
- [ ] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Create a new theme `ToolbarTheme` in `styles.xml` to set it on the toolbar to get the desired color and icons.
- Modify `Tab.java` and `KioskTranslator.java` to always return icons having white color.

#### Fixes the following issue(s)

- Fixes #3042 

### Testing apk
[NewPipe.zip](https://github.com/TeamNewPipe/NewPipe/files/4534541/NewPipe.zip)

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

**Comment by IAmSSH** [comment]
@Stypox 
Request your review please.
Thanks.

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -115,6 +115,97 @@
         <item name="preferenceTheme">@style/PreferenceThemeOverlay.v14.Material</item>
     </style>
 
+    <!-- DrawerArrowStyle is used to change the color of hamburger-icon to white -->
+    <style name="DrawerArrowStyle" parent="Widget.AppCompat.DrawerArrowToggle">
+        <item name="color">@android:color/white</item>
+    </style>
+
+    <style name="ToolbarTheme" parent="ThemeOverlay.AppCompat.ActionBar">
```
Comment (Stypox) [inline app/src/main/res/values/styles.xml:123]: Isn't there a way to inherit all of the icon definitions from the dark theme, so that they don't have to be specified two times?

**Comment by IAmSSH** [comment]
I know its not good practice, I tried doing that, but there's a problem.
If we inherit `DarkTheme`, we essentially also inherit its parent the `Theme.AppCompat.NoActionBar`. There are some styles in `Theme.AppCompat.NoActionBar` that make the text in the spinner that is used to select the quality of a video (in video detail fragment) appear black along with its background, so it becomes unreadable. 
I tried overriding some styles but it didn't work out.
I'll try some more and let you know.

**Comment by IAmSSH** [comment]
I changed the spinner items to always be of red colour, irrespective of the theme selected.

**Comment by Stypox** [comment]
I just realized one thing: maybe there is no need to create a new style in `styles.xml` just for icons, but it is enough to force the white drawables (instead of using `?attr/`) in all menus, since there is only one global style for toolbars. Anyway, the new code looks better, and if my above assumption can't be implemented it is good to go after some testing ;-)

**Comment by IAmSSH** [comment]
I think the hamburger icon's colour cannot be changed without using the `DrawerArrowStyle` style, since it does not belong to our set of style icons (defined in Dark and Light themes), but is preset. So, it might be necessary to keep the style.

**Comment by Stypox** [comment]
Ok, then this is correct approach.

I found some issues (tested on an emulator with Android 4.4 API 19):
- The background color of the toolbar does not change according to the service (i.e. try with soundcloud). It should have the same background color as the tab selector in the main page.
- The background color of the stream selector in the video detail activity should not be red, but should follow the chosen app theme (i.e. dark if dark theme, white if light theme, etc.), just like three-dot menus do.
- The color of the `______` line under the currently selected tab in the main page should always be white, but instead it changes color when changing theme.
- In download activity the color of the "List" toolbar button is black, while it should be white.
- The toolbar of the about activity still has black icons and text when using white theme.

**Comment by IAmSSH** [comment]
Okay, I'll look into it

**Comment by IAmSSH** [comment]
Regarding the last comment:
>The background color of the stream selector in the video detail activity should not be red, but should follow the chosen app theme (i.e. dark if dark theme, white if light theme, etc.), just like three-dot menus do.

You mean 


The menu to select stream quality is red but it should be like the colour of dot menu popup, i.e 


The dot menu popup is always of the colour as shoown in the image irrespective of the stream or the theme, so should I make it similar to that (gray background and white text), or you meant something else?

**Comment by Stypox** [comment]
Oh, that's another issue. Both menus should have dark background with the dark theme and light background with the light theme. I guess you need to do something similar to what you did with `DrawerArrowStyle`

**Comment by IAmSSH** [comment]
The app is working fine now. All issues are resolved.
The only thing remaining is the colour of the spinner in video detail fragment. It does not change according to Light and Dark themes, but is consistent with the colours of different services.
I think it appears fine like this. If you want it to be changed do let me know.

**Comment by Stypox** [comment]
- The "save playlist" toolbar button in the playlist fragment is black in white theme, while it should always be white
- The download dialog (opened from the video detail fragment) has two issues: the stream selector has a red background color, while its background should follow the theme, and the back button (top left) is black in white theme, while it should always be white
- The toolbar in background/popup player activities still has black icons and title in white theme, while they should be white

These two issues you posted screenshots above are still not solved:
- The three-dot menu has a dark background independently of the theme, while it should follow the theme
- The stream selector in video detail fragment has a red background, while it should follow the theme

**Comment by IAmSSH** [comment]
There were some problems with the current approach. Since the Toolbar had the DarkTheme as the parent, so the styles of DarkTheme were getting applied to it statically and they were not changing with the Themes or the Services and it created problems everywhere.

So I had to revert back to the earlier approach that involved copying icons in the ToolbarTheme, it solved all the problems.

**Comment by IAmSSH** [comment]
Hey,
I removed the dead code.
Is there anything else that is remaining?

**Comment by wb9688** [comment]
@IAmSSH: #3271 changed the names of some icons, so you have to change your PR accordingly.

PS: Please stay longer on IRC until we've seen your question and answered, or use e.g. Matrix.

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -115,6 +115,107 @@
...
+        <item name="drawerArrowStyle">@style/DrawerArrowStyle</item>
+        <item name="colorControlNormal">@color/white</item>
+
+        <item name="thumbs_up">@drawable/ic_thumb_up_white_24dp</item>
```
Comment (Stypox) [inline app/src/main/res/values/styles.xml:127]: Now the build is failing because some icon attrs were renamed (e.g. thumbs_up is now ic_thumbs_up). So please copy-paste this list of attributes again

Comment (IAmSSH) [reply app/src/main/res/values/styles.xml]: Done. All checks passing.
Thanks

**Comment by IAmSSH** [comment]
The required changes are done.

PS: @wb9688 Sorry about the IRC. I was really having a hard time figuring out how to work with it. I have set up Matrix now.

File: app/src/main/java/org/schabi/newpipe/MainActivity.java [hunk 3]
```diff
@@ -138,7 +138,15 @@ protected void onCreate(final Bundle savedInstanceState) {
...
         }
 
-        setSupportActionBar(findViewById(R.id.toolbar));
+        Toolbar toolbar = findViewById(R.id.toolbar);
+
+        toolbar.setPopupTheme(
+                ThemeHelper.isLightThemeSelected(this)
+                ? R.style.PopupThemeLight
+                : R.style.PopupThemeDark
+        );
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/MainActivity.java]: Why not use `R.style.ToolbarTheme` here, too?

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 4]
```diff
@@ -743,6 +743,9 @@ private void setupActionBar(final StreamInfo info) {
             public void onItemSelected(final AdapterView<?> parent, final View view,
                                        final int position, final long id) {
                 selectedVideoStreamIndex = position;
+                RelativeLayout r = (RelativeLayout) parent.getChildAt(0);
+                ((TextView) r.getChildAt(1)).setTextColor(getResources().getColor(R.color.white));
+                ((TextView) r.getChildAt(2)).setTextColor(getResources().getColor(R.color.white));
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:748]: Isn't there any other way to do this? It seems to be a workaround to me, though if there is no other way then ok, I guess the toolbar inflated layout is always the same.

File: app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java [hunk 5]
```diff
@@ -471,12 +470,13 @@ private void updateBookmarkButtons() {
...
-                ? R.attr.ic_playlist_add : R.attr.ic_playlist_check;
+                ? R.drawable.ic_playlist_add_white_24dp
+                : R.drawable.ic_playlist_add_check_white_24dp;
 
         final int titleRes = playlistEntity == null
                 ? R.string.bookmark_playlist : R.string.unbookmark_playlist;
 
-        playlistBookmarkButton.setIcon(ThemeHelper.resolveResourceIdFromAttr(activity, iconAttr));
+        playlistBookmarkButton.setIcon(iconAttr);
         playlistBookmarkButton.setTitle(titleRes);
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java:480]: While you are at it, you could move the variables above directly here:
```suggestion
        playlistBookmarkButton.setIcon(playlistEntity == null
                ? R.drawable.ic_playlist_add_white_24dp
                : R.drawable.ic_playlist_add_check_white_24dp);
        playlistBookmarkButton.setTitle(playlistEntity == null
                ? R.string.bookmark_playlist
                : R.string.unbookmark_playlist);
```

File: app/src/main/java/us/shandian/giga/ui/fragment/MissionsFragment.java [hunk 6]
```diff
@@ -224,9 +224,15 @@ private void updateList() {
...
-                    : ThemeHelper.resolveResourceIdFromAttr(requireContext(), R.attr.ic_list));
+            boolean isLight = ThemeHelper.isLightThemeSelected(mContext);
+            int icon;
+
+            if (mLinear)
+                icon = R.drawable.ic_grid_white_24dp;
+            else
+                icon = R.drawable.ic_list_white_24dp;
+
+            mSwitch.setIcon(icon);
```
Comment (Stypox) [inline app/src/main/java/us/shandian/giga/ui/fragment/MissionsFragment.java]: Here, too. Also, `isLight` is unused.
```suggestion
            mSwitch.setIcon(mLinear
                    ? R.drawable.ic_grid_white_24dp
                    : R.drawable.ic_list_white_24dp);
```

File: app/src/main/res/layout/error_retry.xml [hunk 7]
```diff
@@ -27,6 +27,7 @@
         android:textAlignment="center"
         android:textAllCaps="true"
         android:textAppearance="@style/TextAppearance.AppCompat.Body1"
+        android:background="?attr/colorPrimary"
```
Comment (Stypox) [inline app/src/main/res/layout/error_retry.xml:30]: Why was this added? I don't think it is needed after @B0pol introduced `ServiceColoredButton`

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java (hunk 4 above)
Comment (IAmSSH) [reply app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:748]: I couldn't find any direct way of styling just the first element of a spinner, so had to use this approach.

**Comment by IAmSSH** [comment]
Done @Stypox

**Comment by wb9688** [comment]
Should we move this to v0.20.0 seeing how @IAmSSH hasn't replied for 25 days and it not being a bug fix (so not really worth it fixing it ourself imho)?

**Comment by TiA4f8R** [comment]
Closing in favor of #5927.
